service/src/main/java/bio/terra/profile/app/controller/SpendReportingApiController.java

```diff
@@ -0,0 +1,58 @@
+package bio.terra.profile.app.controller;
+
+import bio.terra.common.exception.InconsistentFieldsException;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.common.iam.AuthenticatedUserRequestFactory;
+import bio.terra.profile.api.SpendReportingApi;
+import bio.terra.profile.model.SpendReport;
+import bio.terra.profile.service.spendreporting.SpendReportingService;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Date;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+@Controller
+public class SpendReportingApiController implements SpendReportingApi {
+
+  private final HttpServletRequest request;
+  private final SpendReportingService spendReportingService;
+  private final AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+
+  @Autowired
+  public SpendReportingApiController(
+      HttpServletRequest request,
+      SpendReportingService spendReportingService,
+      AuthenticatedUserRequestFactory authenticatedUserRequestFactory) {
+    this.request = request;
+    this.spendReportingService = spendReportingService;
+    this.authenticatedUserRequestFactory = authenticatedUserRequestFactory;
+  }
+
+  @Override
+  public ResponseEntity<SpendReport> getSpendReport(
+      UUID id,
+      @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) Date spendReportStartDate,
+      @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) Date spendReportEndDate) {
+    OffsetDateTime startDate = spendReportStartDate.toInstant().atOffset(ZoneOffset.UTC);
+    OffsetDateTime endDate = spendReportEndDate.toInstant().atOffset(ZoneOffset.UTC);
+    throwsIfParametersAreNotValid(startDate, endDate);
+
+    AuthenticatedUserRequest authenticatedUserRequest =
+        authenticatedUserRequestFactory.from(request);
+    return new ResponseEntity<>(
+        spendReportingService.getSpendReport(id, startDate, endDate, authenticatedUserRequest),
+        HttpStatus.OK);
+  }
+
+  private void throwsIfParametersAreNotValid(OffsetDateTime startDate, OffsetDateTime endDate) {
```

Comment:
We should guard against bogus dates in the far past and future as well; we might just want to match whatever the azure cost management api allows for now to at least avoid a potential 500.

Reply:
On the Rawls side we did a maxDateRange that is configurable. https://github.com/broadinstitute/rawls/blob/885608b490e9b1418f7edc5622fb7fc3bb716962/core/src/main/scala/org/broadinstitute/dsde/rawls/config/SpendReportingServiceConfig.scala

Reply:
Add configuration and validation for the date range